cleanlab/rank.py

```diff
@@ -427,6 +429,133 @@ def get_label_quality_ensemble_scores(
+        class 0, 1, ..., K-1.
+
+        **Caution**: `pred_probs` from your model must be out-of-sample!
+        You should never provide predictions on the same examples used to train the model,
+        as these will be overfit and unsuitable for finding label-errors.
+        To obtain out-of-sample predicted probabilities for every datapoint in your dataset, you can use :ref:`cross-validation <pred_probs_cross_val>`.
+        Alternatively it is ok if your model was trained on a separate dataset and you are only evaluating
+        data that was previously held-out.
+
+    consensus_method : str or List[str]
+        Describing which algorithm to reach the consensus label.
+        If a List is passed, then the 0th element of List is the method used to produce
+        consensus_label, quality_of_consensus, annotator_agreement (may or may not be affected by this argument).
+        The 1st, 2nd, 3rd, etc. elements of this List are output as extra columns in the returned DF with names formatted as:
+        consensus_label_SUFFIX, quality_of_consensus_SUFFIX
+        where SUFFIX = the str element of this list.
+
+    kwargs : Same additional arguments as for get_label_quality_scores().
+
+    Returns
+    -------
+    df : pandas DataFrame in which each row corresponds to one example, with columns:
+                quality_annotator_1, quality_annotator_2, ..., quality_annotator_M, annotator_agreement, consensus_label, quality_of_consensus
+            (Here _1, _2, ..., _M suffixes may be replaced column names used to describe annotators in `labels` input)
+    """
+
+    # Compute the overall label health score for each annotator (part of annotator_stats)
+    overall_label_health_score_df = labels_multiannotator.apply(
+        overall_label_health_score, args=[pred_probs], verbose=False
+    )
+
+    # Compute the number of labels labeled/annotated by each annotator (part of annotator_stats)
+    num_labeled = labels_multiannotator.count()
+
+    # Find the worst labeled class for each annotator (part of annotator_stats)
+    worst_class = labels_multiannotator.apply(get_worst_class, args=[pred_probs])
+
+    # Create annotator_stats DataFrame from its columns
+    annotator_stats = pd.DataFrame(
+        {
+            "overall_quality": overall_label_health_score_df,
+            "num_labeled": num_labeled,
+            "worst_class": worst_class,
+        }
+    )
+
+    # Compute the label quality scores for each annotators' labels
+    label_quality_scores_multiannotator = labels_multiannotator.apply(
+        get_label_quality_scores, args=[pred_probs], **kwargs
+    )
+
+    # Prefix column name referring to the annotators' label quality scores
+    label_quality_scores_multiannotator = label_quality_scores_multiannotator.add_prefix(
+        "quality_of_"
+    )
+
+    # Compute the consensus_labels
+    # TODO: deal with ties, conditional based on consensus_method, consensus_method can be a List[str], add dawid-skene
```

Comment:
Let's break ties randomly for now.
So if there are 4 labels with an equal number of annotators voting for each one (and no more common label), then we'd just choose one of these 4 at random.

Reply:
have broke ties by picking the one with the highest `pred_probs`